packages/insomnia-components/src/dropdown/dropdown-item.tsx

```diff
@@ -1,20 +1,36 @@
 import React, { PureComponent, ReactNode } from 'react';
 import { autoBindMethodsForReact } from 'class-autobind-decorator';
 import styled, { css } from 'styled-components';
+import classnames from 'classnames';
 
-export interface DropdownItemProps {
+interface ControlledDropdown<T> {
+  value: T;
+  onClick?: (value: T, event: React.MouseEvent<HTMLButtonElement>) => void | Promise<void>;
+}
+
+interface UncontrolledDropdown {
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+}
+
+const isControlledInput = <T extends unknown>(dropdown: DropdownItemProps<T>): dropdown is ControlledDropdown<T> => (
+  Object.prototype.hasOwnProperty.call(dropdown, 'value')
+);
+
+const isUncontrolledInput = <T extends unknown>(dropdown: DropdownItemProps<T>): dropdown is ControlledDropdown<T> => (
```

Comment:
Should this be the following? 
```suggestion
const isUncontrolledInput = <T extends unknown>(dropdown: DropdownItemProps<T>): dropdown is UncontrolledDropdown => (
```

Reply:
YEP. heh.

YOU THINK YOU KNOW A GUY (the guy being this code)..

Reply:
fixed in 5d800a3dd5dd111196cd80708b4e1a789e5c7a91